lib/codecs/src/encoding/format/protobuf.rs

```diff
@@ -96,6 +96,8 @@ fn convert_value_raw(
         }
         (Value::Float(f), Kind::Double) => Ok(prost_reflect::Value::F64(f.into_inner())),
         (Value::Float(f), Kind::Float) => Ok(prost_reflect::Value::F32(f.into_inner() as f32)),
+        (Value::Integer(i), Kind::Double) => Ok(prost_reflect::Value::F64(i as f64)),
+        (Value::Integer(i), Kind::Float) => Ok(prost_reflect::Value::F32(i as f32)),
```

Comment:
Even it's a "high precision conversion", the farther you get from `0` the more problematic it becomes. I am not fully convinced we want to introduce this. Is the motivation to avoid using a remap transform? (that might also have problems with very large integers)

Perhaps if we document very clearly the caveats especially when it comes to large `i64` we can justify introducing this for convenience. Users who know their data can leverage it.

Reply:
@pront could you clarify "the farther you get from `0` the more problematic it becomes"?  I added tests in this PR specifically to see if that was the case.

The motivation is to avoid using a remap to transform, particularly when the conversion to integer is the result of another parser and not a `remap` itself, e.g. the JSON parser deciding that the number it parsed is an integer instead of a float.

Reply:
I see now, this use case makes sense.

Reply:
> The problem is that integer values larger than the number of mantissa bits will have the low-order bits cleared.

This is what I was referring to as problematic. These conversions are lossy and that would need to be documented. VRL also performs lossy conversions, see [here](https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoidmFsdWUgPSBcIjkyMjMzNzIwMzY4NTQ3NzU4MDhcIlxuLmEgPSBwYXJzZV9qc29uISh2YWx1ZSlcblxudmFsdWUgPSBcIjkyMjMzNzIwMzY4NTQ3NzU4MDhcIiAqIDJcbi5iID0gcGFyc2VfanNvbiEodmFsdWUpIiwiZXZlbnQiOnt9LCJpc19qc29ubCI6ZmFsc2UsImVycm9yIjpudWxsfQ%3D%3D).